[release][ci] air_benchmark_tune_torch_mnist failed

### What happened + What you expected to happen

[Build ](https://buildkite.com/ray-project/release-tests-branch/builds/1080#0183906e-87c7-4d03-835f-181c85c547ed)
[Cluster](https://console.anyscale-staging.com/o/anyscale-internal/projects/prj_qC3ZfndQWYYjx2cz8KWGNUL4/clusters/ses_TAmfcLbZGWWMu32PL5XNg1nY)

```
Results run 3: {'train_time': 224.828414328, 'tune_time': 348.08502680400034}
Full results: {'train_times': [226.341469459, 333.7736055319999, 224.828414328], 'train_mean': 261.64782977299996, 'train_sd': 51.004365699323685, 'tune_times': [343.143759339, 345.24248954199993, 348.08502680400034], 'tune_mean': 345.49042522833344, 'tune_sd': 2.024867916447225}
Traceback (most recent call last):
  File "workloads/tune_torch_benchmark.py", line 152, in <module>
    main()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "workloads/tune_torch_benchmark.py", line 144, in main
    ), f"{mean_tune_time:.2f} > {threshold:.2f} = {factor:.1f} * {mean_train_time:.2f}"
AssertionError: 345.49 > 313.98 = 1.2 * 261.65
```

### Versions / Dependencies

NA

### Reproduction script

NA

### Issue Severity

_No response_

--- a/release/air_tests/air_benchmarks/workloads/tune_torch_benchmark.py
+++ b/release/air_tests/air_benchmarks/workloads/tune_torch_benchmark.py
@@ -36,6 +36,13 @@ def get_trainer(num_workers: int = 4, use_gpu: bool = False):
     def train_loop(config):
         train_func(use_ray=True, config=config)
 
+    # We are using STRICT_PACK here to do an apples to apples comparison.
+    # PyTorch defaults to using multithreading, so if the workers are spread,
+    # they are able to utilize more resources. We would effectively be comparing
+    # X tune runs with 2 CPUs per worker vs. 1 tune run with up to 8 CPUs per
+    # worker. Using STRICT_PACK avoids this by forcing all workers to be
+    # co-located.
+
     trainer = TorchTrainer(
         train_loop_per_worker=train_loop,
         train_loop_config=CONFIG,
@@ -44,6 +51,7 @@ def get_trainer(num_workers: int = 4, use_gpu: bool = False):
             resources_per_worker={"CPU": 2},
             trainer_resources={"CPU": 0},
             use_gpu=use_gpu,
+            placement_strategy="STRICT_PACK",
         ),
     )
     return trainer

